### ISSUE TITLE
Nginx - epub download works, pdf doesn't

### ISSUE THREAD

**Issue by LunkSnee** [seblucas/cops#456]
I have COPS working fine, I thought. EPub files download great. Multiple databases all working.
I use nginx and a subdirectory: https://domain/books/.
However, I now added some PDFs to my library, and while the epub files continue to work, the PDFs don't. They are in the same library and same directory structure as the epub files.

Here are some of the errors I see when I try to download a PDF book:

`2019/08/22 21:58:21 [error] 28969#28969: *19764 FastCGI sent in stderr: "PHP message: PHP Warning:  fopen(/books/Humble/Patrick Merrell/AHA! 125 Original &amp; Amusing Word Pu (304)/AHA! 125 Original &amp; Amusing Wor - Patrick Merrell.pdf): failed to open stream: No such file or directory in /var/www/books/fetch.php on line 121`
`PHP message: PHP Warning:  filesize(): stat failed for /books/Humble/Patrick Merrell/AHA! 125 Original &amp; Amusing Word Pu (304)/AHA! 125 Original &amp; Amusing Wor - Patrick Merrell.pdf in /var/www/books/fetch.php on line 122`
`PHP message: PHP Warning:  fpassthru() expects parameter 1 to be resource, boolean given in /var/www/books/fetch.php on line 123" while reading response header from upstream, client: XXXXXXXXX, server: XXXXXXXX, request: "GET /books/fetch.php?data=41&type=pdf&id=304&db=2 HTTP/2.0", upstream: "fastcgi://unix:/var/run/php/php7.2-fpm.sock:", host: "XXXXXXXXXX", referrer: "https://XXXXXXXXX/books/index.php?page=4&db=2"`

Any help? The fact that it is only PDF with the problem is so strange.

**Comment by woidi** [comment]
epub files receive some special handling in COPS, but PDFs should also work.
Somehow COPS is not putting the calibre_directory in front of the filename; can you provide the links to the epub and the pdf of a single book as seen in a web browser? (right click on the EPUB and PDF buttons and "copy link address".) You can replace your server/domainname of course.

**Comment by LunkSnee** [comment]
Thanks for the question.
I either have a PDF version or ePub version of each book, so unfortunately I can't easily show you the same book. I can tell you the PDFs are present and readable exactly like the ePubs. Two of the links:
```
https://[domain.ext]/books/download/66/2/The%20Ape%27s%20Wife%20and%20Other%20Storie%20-%20Caitlin%20R.%20Kiernan.epub
https://[domain.ext]/books/download/125/2/The%20Astronaut%27s%20Cookbook%20-%20Charles%20Bourland.pdf
```
Turns out I fixed it by adding the complete path as the 'calibre_internal_directory' rather than using the location parameter in my webserver configuration. 

What got me was that epubs worked but PDFs wouldn't. Now both do, with what I suppose is the right 'calibre_internal_directory' setting.
